### PR TITLE
Openclaw迁移功能（时间简史）初步方案

### PR DESCRIPTION
**Pull request by Jacobinwwey**
## Motivation / 动机
在从 OpenClaw 迁移到 AstrBot 的实际场景中，目前缺少一个官方、可审计、可回放的迁移入口。用户通常需要手工拷贝目录和解析历史记忆，存在三类问题：

迁移流程不可预览：执行前无法确认会迁移哪些内容、规模多大。
迁移结果不可追溯：缺少统一产物与迁移摘要，不利于排障和二次校验。
配置格式不兼容：OpenClaw 常见 JSON 配置无法直接转换为结构化 TOML 快照。
This PR introduces a preliminary OpenClaw migration workflow for AstrBot CLI with dry-run preview, artifactized outputs, and a generated “time brief history”.
本 PR 引入 OpenClaw 迁移的 CLI 初步方案，支持 dry-run 预览、迁移产物落盘和“时间简史”自动生成。

## Key Changes / 核心改动
OpenClaw migration CLI entry
新增 `astrbot migrate openclaw` 命令，支持：
- `--dry-run`：只读预览，不写入。
- `--source <path>`：自定义 OpenClaw 根目录（默认 `~/.openclaw`）。
- `--target <path>`：自定义迁移输出目录。

Memory migration candidate collection (sqlite + markdown)
读取 OpenClaw 工作区中的：
- `memory/brain.db`（`memories` 表，兼容 `content/value/text/memory` 等列别名）
- `MEMORY.md`
- `memory/*.md`
并做去重，形成标准化 memory entries。

Workspace artifact snapshot
递归收集并复制 OpenClaw 工作区文件到迁移目录，形成可审计快照。

Config JSON -> TOML snapshot conversion
检测 OpenClaw 侧 `config.json/settings.json`，输出：
- `config.original.json`
- `config.migrated.toml`
用于后续人工校验与渐进接入。

Generated timeline artifact (Time Brief History)
新增 `time_brief_history.md`：按时间顺序生成迁移记忆摘要，作为“时间简史（初步方案）”产物，便于人工复盘。

Migration summary report
输出 `migration_summary.json`，包含候选条目数量、来源分布、工作区文件规模、配置检测状态、写入结果等。

## Main Files / 主要文件
astrbot/cli/commands/cmd_migrate.py
astrbot/cli/commands/__init__.py
astrbot/cli/__main__.py
tests/unit/test_cli_openclaw_migrate.py
docs/zh/deploy/astrbot/cli.md
docs/en/deploy/astrbot/cli.md

## Config Highlights / 配置要点（默认值）
本 PR 不引入新的全局 `provider_settings` 配置项；通过 CLI 参数控制行为：
- `astrbot migrate openclaw --dry-run`
- `astrbot migrate openclaw --source <path>`
- `astrbot migrate openclaw --target <path>`

默认输入/输出约定：
- Source: `~/.openclaw`
- Output: `data/migrations/openclaw/run-<timestamp>`

## Compatibility / 兼容性
This is NOT a breaking change.
该改动为新增 CLI 子命令，不影响现有运行路径与配置语义。

## Verification / 验证
Verification command 1:

`uv run ruff check astrbot/cli/commands/cmd_migrate.py astrbot/cli/commands/__init__.py astrbot/cli/__main__.py tests/unit/test_cli_openclaw_migrate.py`

Output: `All checks passed!`

Verification command 2:

`uv run pytest -q tests/unit/test_cli_openclaw_migrate.py`

Output: `3 passed in 0.11s`

Verification command 3:

`uv run astrbot help migrate`

Checks include:
- CLI command registration (`migrate -> openclaw`)
- Help text visibility

## Summary by Sourcery

Add a new CLI-based OpenClaw-to-AstrBot migration workflow with preview and artifact generation capabilities.

New Features:
- Introduce `astrbot migrate openclaw` command to migrate OpenClaw workspaces into AstrBot migration artifacts with optional dry-run and configurable source/target paths.
- Support extraction and de-duplication of memory entries from OpenClaw sqlite and markdown sources into a normalized JSONL format, along with a generated time-ordered history markdown file.
- Add conversion of OpenClaw JSON configuration into TOML snapshot files alongside preserved original JSON for manual inspection and gradual adoption.

Documentation:
- Document the new OpenClaw migration command and default input/output locations in both English and Chinese CLI deployment guides.

Tests:
- Add unit tests covering OpenClaw sqlite schema compatibility, markdown parsing and de-duplication behavior, artifact writing, TOML conversion, and invalid config error handling for the migration workflow.

## Summary by Sourcery

Add a new CLI-based OpenClaw-to-AstrBot migration workflow with dry-run support and artifact generation, and document and test the new migration command.

New Features:
- Introduce a `migrate` CLI group with an `openclaw` subcommand to migrate OpenClaw workspaces into AstrBot migration artifacts with configurable source/target paths and dry-run mode.
- Collect and normalize memory entries from OpenClaw sqlite and markdown sources into a deduplicated JSONL dataset plus a generated time-ordered timeline markdown file.
- Snapshot the OpenClaw workspace files and convert JSON configuration into TOML snapshots alongside the preserved original JSON, emitting a structured migration summary.

Documentation:
- Document the OpenClaw migration command, options, and default source/target locations in both English and Chinese CLI deployment guides.

Tests:
- Add unit tests for the OpenClaw migration CLI command, sqlite/markdown memory collection and deduplication, workspace snapshot behavior, TOML conversion edge cases, and error handling for invalid inputs.

### CODE REVIEW

**Comment by Jacobinwwey** [comment]
@sourcery-ai review

**Comment by Jacobinwwey** [comment]
@sourcery-ai review

**Comment by Jacobinwwey** [comment]
@sourcery-ai review

**Comment by Jacobinwwey** [comment]
@sourcery-ai review

**Comment by Jacobinwwey** [comment]
@sourcery-ai review

File: astrbot/cli/utils/openclaw_migrate.py [hunk 1]
```diff
@@ -0,0 +1,536 @@
...
+    try:
+        ts = float(raw)
+        if ts > 1e12:
+            ts /= 1000.0
+        return dt.datetime.fromtimestamp(ts, tz=dt.timezone.utc).isoformat()
+    except Exception:
+        return None
+
+
+def _normalize_timestamp(raw: Any) -> str | None:
```
Comment (SourceryAI) [inline astrbot/cli/utils/openclaw_migrate.py]: **issue (complexity):** Consider extracting timestamp normalization and markdown parsing into smaller helper functions to make the control flow easier to follow without changing behavior.

A couple of localized refactors would reduce cognitive load without changing behavior:

### 1. Make timestamp handling more linear

`_normalize_timestamp` currently interleaves numeric/epoch and string/ISO logic. Splitting these paths into small helpers makes the decision tree clearer while preserving all existing behavior:

```python
def _normalize_timestamp(raw: Any) -> str | None:
    if raw is None:
        return None

    if isinstance(raw, (int, float)):
        return _normalize_numeric_timestamp(raw)

    text = str(raw).strip()
    if not text:
        return None

    if text.isdigit():
        return _normalize_numeric_timestamp(text)

    return _normalize_iso_like_timestamp(text)


def _normalize_numeric_timestamp(raw: float | int | str) -> str:
    normalized = _timestamp_from_epoch(raw)
    return normalized if normalized is not None else str(raw)


def _normalize_iso_like_timestamp(text: str) -> str:
    maybe_iso = text.replace("Z", "+00:00")
    try:
        parsed = dt.datetime.fromisoformat(maybe_iso)
        if parsed.tzinfo is None:
            parsed = parsed.replace(tzinfo=dt.timezone.utc)
        return parsed.isoformat()
    except Exception:
        return text
```

This keeps the same semantics (including “fallback to original text”) but makes it obvious what formats are intended to be handled.

### 2. Flatten markdown parsing into clearly separated steps

`_parse_markdown_file` currently mixes line iteration, bullet stripping, structured detection, and key/body construction. Extracting “line normalization” and “entry building” makes the main loop easier to scan:

```python
def _normalize_markdown_line(raw_line: str) -> str:
    stripped = raw_line.strip()
    if not stripped or stripped.startswith("#"):
        return ""
    return stripped[2:] if stripped.startswith("- ") else stripped


def _build_markdown_entry(
    line: str,
    stem: str,
    line_no: int,
    default_category: str,
    mtime: str | None,
    source: str,
    order: int,
) -> MemoryEntry | None:
    parsed = _parse_structured_line(line)
    if parsed is not None:
        key, text = parsed
        key = _normalize_key(key, line_no)
        body = text.strip()
    else:
        key = f"openclaw_{stem}_{line_no}"
        body = line.strip()

    if not body:
        return None

    return MemoryEntry(
        key=key,
        content=body,
        category=default_category,
        timestamp=mtime,
        source=source,
        order=order,
    )
```

Then `_parse_markdown_file` becomes a simple driver:

```python
def _parse_markdown_file(
    path: Path, default_category: str, stem: str, order_offset: int
) -> list[MemoryEntry]:
    content = path.read_text(encoding="utf-8", errors="replace")
    mtime = _normalize_timestamp(path.stat().st_mtime)

    entries: list[MemoryEntry] = []
    for line_no, raw_line in enumerate(content.splitlines(), start=1):
        line = _normalize_markdown_line(raw_line)
        if not line:
            continue

        entry = _build_markdown_entry(
            line=line,
            stem=stem,
            line_no=line_no,
            default_category=default_category,
            mtime=mtime,
            source=f"markdown:{path}",
            order=order_offset + len(entries),
        )
        if entry is not None:
            entries.append(entry)

    return entries
```

Functionality stays identical (including headers/bullets handling and key generation), but responsibilities per function are tighter, which should address part of the “mini-DSL” complexity concern without changing behavior or file layout.

✅ Addressed in 8f0a9753715ec7f31629b1b0acacfcd4584dcfdf: Timestamp normalization has been split with `_timestamp_from_epoch`, and markdown parsing is structured around `_parse_markdown_file` and helpers, making the logic more linear while preserving semantics.

File: astrbot/cli/utils/openclaw_toml.py [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+        return "[" + ", ".join(_toml_literal(v) for v in value) + "]"
+    if isinstance(value, dict):
+        pairs = ", ".join(
+            f"{_toml_format_key(str(k))} = {_toml_literal(v)}" for k, v in value.items()
+        )
+        return "{ " + pairs + " }"
+    return _toml_quote(str(value))
+
+
+def json_to_toml(data: dict[str, Any]) -> str:
```
Comment (SourceryAI) [inline astrbot/cli/utils/openclaw_toml.py:72]: **issue (complexity):** Consider refactoring the serializer to separate null-handling policy and value classification from formatting logic to make the TOML conversion code clearer and easier to maintain.

You can keep the same behavior but reduce the “mini-serializer” feel by (1) separating policy from formatting, and (2) making the table traversal more declarative.

### 1. Extract the NULL policy out of `_toml_literal`

Right now `_toml_literal` both formats values and encodes the “None → `NULL_SENTINEL`” policy, which makes it harder to see what TOML types are actually supported.

You can normalize the input once at the top of `json_to_toml`, then let `_toml_literal` only deal with TOML-compatible values:

```python
def _normalize_nulls(value: Any) -> Any:
    if value is None:
        return NULL_SENTINEL
    if isinstance(value, list):
        return [_normalize_nulls(v) for v in value]
    if isinstance(value, dict):
        return {k: _normalize_nulls(v) for k, v in value.items()}
    return value
```

Use it at the beginning of `json_to_toml`:

```python
def json_to_toml(data: dict[str, Any]) -> str:
    data = _normalize_nulls(data)
    lines: list[str] = []

    def emit_table(obj: dict[str, Any], path: list[str]) -> None:
        ...
```

Then `_toml_literal` no longer needs to know about `None`:

```python
def _toml_literal(value: Any) -> str:
    if isinstance(value, bool):
        return "true" if value else "false"
    if isinstance(value, int):
        return str(value)
    if isinstance(value, float):
        if not math.isfinite(value):
            raise ValueError(f"non-finite float value is not TOML-compatible: {value}")
        return repr(value)
    if isinstance(value, str):
        return _toml_quote(value)
    if isinstance(value, list):
        return "[" + ", ".join(_toml_literal(v) for v in value) + "]"
    if isinstance(value, dict):
        pairs = ", ".join(
            f"{_toml_format_key(str(k))} = {_toml_literal(v)}" for k, v in value.items()
        )
        return "{ " + pairs + " }"
    return _toml_quote(str(value))
```

This keeps output identical (since `None` becomes the string `__NULL__` before formatting) but makes the serializer’s responsibilities narrower and clearer.

### 2. Make value classification in `emit_table` explicit

The current `emit_table` loop embeds classification logic inline. Introducing a tiny helper makes it easier to read and adjust later (e.g., if you ever tweak the array-of-tables rule) without changing behavior.

```python
from enum import Enum, auto


class _TomlValueKind(Enum):
    SCALAR = auto()
    TABLE = auto()
    ARRAY_OF_TABLES = auto()


def _classify_toml_value(value: Any) -> _TomlValueKind:
    if isinstance(value, dict):
        return _TomlValueKind.TABLE
    if isinstance(value, list) and value and all(isinstance(item, dict) for item in value):
        return _TomlValueKind.ARRAY_OF_TABLES
    return _TomlValueKind.SCALAR
```

Use it inside `emit_table`:

```python
def emit_table(obj: dict[str, Any], path: list[str]) -> None:
    scalar_items: list[tuple[str, Any]] = []
    nested_dicts: list[tuple[str, dict[str, Any]]] = []
    array_tables: list[tuple[str, list[dict[str, Any]]]] = []

    for key, value in obj.items():
        kind = _classify_toml_value(value)
        skey = str(key)
        if kind == _TomlValueKind.TABLE:
            nested_dicts.append((skey, value))  # type: ignore[arg-type]
        elif kind == _TomlValueKind.ARRAY_OF_TABLES:
            array_tables.append((skey, value))  # type: ignore[arg-type]
        else:
            scalar_items.append((skey, value))

    ...
```

The behavior (including the “only non-empty `list[dict]` is array-of-tables; empty lists stay scalars”) stays exactly the same, but the main traversal is more declarative and easier to maintain.

✅ Addressed in 8f0a9753715ec7f31629b1b0acacfcd4584dcfdf: The TOML conversion code now separates value classification via `_classify_items` and constrains `_toml_literal` to formatting concerns, with explicit handling of nulls and non-finite floats.

**Review by SourceryAI — COMMENTED** [review]
Hey - I've found 2 issues, and left some high level feedback:

- In `cmd_migrate.migrate_openclaw`, `click.Path(..., resolve_path=True)` means `target_path` will always be absolute, so the `if target_dir.is_absolute()` branch in `run_openclaw_migration` is effectively dead; either drop `resolve_path=True` for `--target` (and possibly `--source`) to preserve the documented AstrBot-root-relative semantics, or simplify `run_openclaw_migration` to treat the paths as already fully resolved.
- The OpenClaw migration internals (`MemoryEntry`, `MigrationReport`, `run_openclaw_migration`) are re-exported from `astrbot.cli.utils.__init__`, which broadens the public CLI utils surface; consider keeping them in the dedicated `openclaw_migrate` module (or under a more explicitly experimental namespace) to avoid committing to this API shape long term.

<details>
<summary>Prompt for AI Agents</summary>

~~~markdown
Please address the comments from this code review:

## Overall Comments
- In `cmd_migrate.migrate_openclaw`, `click.Path(..., resolve_path=True)` means `target_path` will always be absolute, so the `if target_dir.is_absolute()` branch in `run_openclaw_migration` is effectively dead; either drop `resolve_path=True` for `--target` (and possibly `--source`) to preserve the documented AstrBot-root-relative semantics, or simplify `run_openclaw_migration` to treat the paths as already fully resolved.
- The OpenClaw migration internals (`MemoryEntry`, `MigrationReport`, `run_openclaw_migration`) are re-exported from `astrbot.cli.utils.__init__`, which broadens the public CLI utils surface; consider keeping them in the dedicated `openclaw_migrate` module (or under a more explicitly experimental namespace) to avoid committing to this API shape long term.

## Individual Comments

### Comment 1
<location path="astrbot/cli/utils/openclaw_migrate.py" line_range="67" />
<code_context>
+        return None
+
+
+def _normalize_timestamp(raw: Any) -> str | None:
+    if raw is None:
+        return None
</code_context>
<issue_to_address>
**issue (complexity):** Consider extracting timestamp normalization and markdown parsing into smaller helper functions to make the control flow easier to follow without changing behavior.

A couple of localized refactors would reduce cognitive load without changing behavior:

### 1. Make timestamp handling more linear

`_normalize_timestamp` currently interleaves numeric/epoch and string/ISO logic. Splitting these paths into small helpers makes the decision tree clearer while preserving all existing behavior:

```python
def _normalize_timestamp(raw: Any) -> str | None:
    if raw is None:
        return None

    if isinstance(raw, (int, float)):
        return _normalize_numeric_timestamp(raw)

    text = str(raw).strip()
    if not text:
        return None

    if text.isdigit():
        return _normalize_numeric_timestamp(text)

    return _normalize_iso_like_timestamp(text)


def _normalize_numeric_timestamp(raw: float | int | str) -> str:
    normalized = _timestamp_from_epoch(raw)
    return normalized if normalized is not None else str(raw)


def _normalize_iso_like_timestamp(text: str) -> str:
    maybe_iso = text.replace("Z", "+00:00")
    try:
        parsed = dt.datetime.fromisoformat(maybe_iso)
        if parsed.tzinfo is None:
            parsed = parsed.replace(tzinfo=dt.timezone.utc)
        return parsed.isoformat()
    except Exception:
        return text
```

This keeps the same semantics (including “fallback to original text”) but makes it obvious what formats are intended to be handled.

### 2. Flatten markdown parsing into clearly separated steps

`_parse_markdown_file` currently mixes line iteration, bullet stripping, structured detection, and key/body construction. Extracting “line normalization” and “entry building” makes the main loop easier to scan:

```python
def _normalize_markdown_line(raw_line: str) -> str:
    stripped = raw_line.strip()
    if not stripped or stripped.startswith("#"):
        return ""
    return stripped[2:] if stripped.startswith("- ") else stripped


def _build_markdown_entry(
    line: str,
    stem: str,
    line_no: int,
    default_category: str,
    mtime: str | None,
    source: str,
    order: int,
) -> MemoryEntry | None:
    parsed = _parse_structured_line(line)
    if parsed is not None:
        key, text = parsed
        key = _normalize_key(key, line_no)
        body = text.strip()
    else:
        key = f"openclaw_{stem}_{line_no}"
        body = line.strip()

    if not body:
        return None

    return MemoryEntry(
        key=key,
        content=body,
        category=default_category,
        timestamp=mtime,
        source=source,
        order=order,
    )
```

Then `_parse_markdown_file` becomes a simple driver:

```python
def _parse_markdown_file(
    path: Path, default_category: str, stem: str, order_offset: int
) -> list[MemoryEntry]:
    content = path.read_text(encoding="utf-8", errors="replace")
    mtime = _normalize_timestamp(path.stat().st_mtime)

    entries: list[MemoryEntry] = []
    for line_no, raw_line in enumerate(content.splitlines(), start=1):
        line = _normalize_markdown_line(raw_line)
        if not line:
            continue

        entry = _build_markdown_entry(
            line=line,
            stem=stem,
            line_no=line_no,
            default_category=default_category,
            mtime=mtime,
            source=f"markdown:{path}",
            order=order_offset + len(entries),
        )
        if entry is not None:
            entries.append(entry)

    return entries
```

Functionality stays identical (including headers/bullets handling and key generation), but responsibilities per function are tighter, which should address part of the “mini-DSL” complexity concern without changing behavior or file layout.
</issue_to_address>

### Comment 2
<location path="astrbot/cli/utils/openclaw_toml.py" line_range="51" />
<code_context>
+    return _toml_quote(str(value))
+
+
+def json_to_toml(data: dict[str, Any]) -> str:
+    """Serialize a JSON-like dict to TOML text used by migration snapshots.
+
</code_context>
<issue_to_address>
**issue (complexity):** Consider refactoring the serializer to separate null-handling policy and value classification from formatting logic to make the TOML conversion code clearer and easier to maintain.

You can keep the same behavior but reduce the “mini-serializer” feel by (1) separating policy from formatting, and (2) making the table traversal more declarative.

### 1. Extract the NULL policy out of `_toml_literal`

Right now `_toml_literal` both formats values and encodes the “None → `NULL_SENTINEL`” policy, which makes it harder to see what TOML types are actually supported.

You can normalize the input once at the top of `json_to_toml`, then let `_toml_literal` only deal with TOML-compatible values:

```python
def _normalize_nulls(value: Any) -> Any:
    if value is None:
        return NULL_SENTINEL
    if isinstance(value, list):
        return [_normalize_nulls(v) for v in value]
    if isinstance(value, dict):
        return {k: _normalize_nulls(v) for k, v in value.items()}
    return value
```

Use it at the beginning of `json_to_toml`:

```python
def json_to_toml(data: dict[str, Any]) -> str:
    data = _normalize_nulls(data)
    lines: list[str] = []

    def emit_table(obj: dict[str, Any], path: list[str]) -> None:
        ...
```

Then `_toml_literal` no longer needs to know about `None`:

```python
def _toml_literal(value: Any) -> str:
    if isinstance(value, bool):
        return "true" if value else "false"
    if isinstance(value, int):
        return str(value)
    if isinstance(value, float):
        if not math.isfinite(value):
            raise ValueError(f"non-finite float value is not TOML-compatible: {value}")
        return repr(value)
    if isinstance(value, str):
        return _toml_quote(value)
    if isinstance(value, list):
        return "[" + ", ".join(_toml_literal(v) for v in value) + "]"
    if isinstance(value, dict):
        pairs = ", ".join(
            f"{_toml_format_key(str(k))} = {_toml_literal(v)}" for k, v in value.items()
        )
        return "{ " + pairs + " }"
    return _toml_quote(str(value))
```

This keeps output identical (since `None` becomes the string `__NULL__` before formatting) but makes the serializer’s responsibilities narrower and clearer.

### 2. Make value classification in `emit_table` explicit

The current `emit_table` loop embeds classification logic inline. Introducing a tiny helper makes it easier to read and adjust later (e.g., if you ever tweak the array-of-tables rule) without changing behavior.

```python
from enum import Enum, auto


class _TomlValueKind(Enum):
    SCALAR = auto()
    TABLE = auto()
    ARRAY_OF_TABLES = auto()


def _classify_toml_value(value: Any) -> _TomlValueKind:
    if isinstance(value, dict):
        return _TomlValueKind.TABLE
    if isinstance(value, list) and value and all(isinstance(item, dict) for item in value):
        return _TomlValueKind.ARRAY_OF_TABLES
    return _TomlValueKind.SCALAR
```

Use it inside `emit_table`:

```python
def emit_table(obj: dict[str, Any], path: list[str]) -> None:
    scalar_items: list[tuple[str, Any]] = []
    nested_dicts: list[tuple[str, dict[str, Any]]] = []
    array_tables: list[tuple[str, list[dict[str, Any]]]] = []

    for key, value in obj.items():
        kind = _classify_toml_value(value)
        skey = str(key)
        if kind == _TomlValueKind.TABLE:
            nested_dicts.append((skey, value))  # type: ignore[arg-type]
        elif kind == _TomlValueKind.ARRAY_OF_TABLES:
            array_tables.append((skey, value))  # type: ignore[arg-type]
        else:
            scalar_items.append((skey, value))

    ...
```

The behavior (including the “only non-empty `list[dict]` is array-of-tables; empty lists stay scalars”) stays exactly the same, but the main traversal is more declarative and easier to maintain.
</issue_to_address>
~~~

</details>

***

Hi @Jacobinwwey! 👋

Thanks for trying out Sourcery by commenting with `@sourcery-ai review`! 🚀

Install the [sourcery-ai bot](https://github.com/apps/sourcery-ai/installations/new) to get automatic code reviews on every pull request ✨

<sub>
Help me be more useful! Please click 👍 or 👎 on each comment and I'll use the feedback to improve your reviews.
</sub>

**Comment by Jacobinwwey** [comment]
@sourcery-ai review

**Comment by Jacobinwwey** [comment]
@sourcery-ai review

**Comment by Jacobinwwey** [comment]
@sourcery-ai review

**Comment by Jacobinwwey** [comment]
@sourcery-ai review

**Comment by Jacobinwwey** [comment]
@sourcery-ai review

**Comment by Jacobinwwey** [comment]
@sourcery-ai review

**Comment by Jacobinwwey** [comment]
@sourcery-ai review

**Review by SourceryAI — COMMENTED** [review]
Hey - I've reviewed your changes and they look great!

***

Hi @Jacobinwwey! 👋

Thanks for trying out Sourcery by commenting with `@sourcery-ai review`! 🚀

Install the [sourcery-ai bot](https://github.com/apps/sourcery-ai/installations/new) to get automatic code reviews on every pull request ✨

<sub>
Help me be more useful! Please click 👍 or 👎 on each comment and I'll use the feedback to improve your reviews.
</sub>

**Comment by Jacobinwwey** [comment]
@sourcery-ai review

**Comment by Jacobinwwey** [comment]
@LIghtJUNction  @Raven95676 [Passed the review by Sourcery]

**Comment by LIghtJUNction** [comment]
用ruff格式化一下

**Comment by Jacobinwwey** [comment]
CI 失败原因已定位：当前唯一失败项是 `Code Format Check / format-check`。我在分支本地按工作流命令复现后，确认是 `uv run ruff format --check .` 失败，涉及 3 个文件的纯格式漂移：`astrbot/cli/utils/openclaw_artifacts.py`、`astrbot/cli/utils/openclaw_memory.py`、`astrbot/cli/utils/openclaw_toml.py`。`uv run ruff check .` 本地是通过的。

我已经提交并推送纯格式修正提交 `8f0a9753`。本地复验结果：
- `uv run ruff format --check .` 通过
- `uv run ruff check .` 通过

现在等待 GitHub Actions 基于最新提交重跑。

@LIghtJUNction 可review
